# Predicting Score based on Number of study hours.

### This is task-01 for the GRIP internship at The Sparks Foundation. Here, we will simply use logistic regression to predict the score based upon no. of hours studied. 

###### Importing required libraries

In [10]:
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

##### Importing the Dataset and reading it

In [7]:
url = 'http://bit.ly/w-data'
df = pd.read_csv(url)

In [8]:
df

Hours  Scores
0     2.5      21
1     5.1      47
2     3.2      27
3     8.5      75
4     3.5      30
5     1.5      20
6     9.2      88
7     5.5      60
8     8.3      81
9     2.7      25
10    7.7      85
11    5.9      62
12    4.5      41
13    3.3      42
14    1.1      17
15    8.9      95
16    2.5      30
17    1.9      24
18    6.1      67
19    7.4      69
20    2.7      30
21    4.8      54
22    3.8      35
23    6.9      76
24    7.8      86

### The Data used to train the model cannot be used to test it so we need to divide the dataset into two parts. 

##### Splitting the Dataset into training and testing sets.

In [38]:
xtrain,xtest,ytrain,ytest = train_test_split(df[['Hours']],df[['Scores']], test_size=0.4)

In [39]:
xtrain

Hours
2     3.2
19    7.4
8     8.3
11    5.9
24    7.8
23    6.9
20    2.7
13    3.3
14    1.1
3     8.5
21    4.8
7     5.5
18    6.1
10    7.7
9     2.7

In [40]:
ytrain

Scores
2       27
19      69
8       81
11      62
24      86
23      76
20      30
13      42
14      17
3       75
21      54
7       60
18      67
10      85
9       25

### Now we will make the model and train it using the training data.

##### Making the Linear Regression model and training it

In [41]:
classifier = LinearRegression()

In [42]:
classifier.fit(xtrain,ytrain)

LinearRegression()

### Next step is to check the model with our testing data. Here, we will check the model by using the Hours data from our testing dataset and compare it with Score data of the testing dataset.

##### Predicting the score based upon hours

In [43]:
classifier.predict(xtest)

array([[47.89021043],
       [53.62549558],
       [89.94896818],
       [41.19904442],
       [38.33140185],
       [28.77259327],
       [28.77259327],
       [92.81661076],
       [19.21378469],
       [23.03730812]])

##### Displaying the actual values of Score

In [44]:
ytest

Scores
12      41
1       47
15      95
22      35
4       30
16      30
0       21
6       88
5       20
17      24

##### Comparing the accuracy of the model's outcome with the actual values of Score

In [45]:
classifier.score(xtest,ytest)

0.9522766777767027

### Clearly, we got a 95.2% accuracy with the help of model which is great.